telethon/telegram_client.py

```diff
@@ -2470,4 +2471,75 @@ def get_input_entity(self, peer):
             'Make sure you have encountered this peer before.'.format(peer)
         )
 
+    def edit_2FA(self, current_password=None, new_password=None, hint='', email=None):
+        """
+        Changes the 2FA settings of the logged in user, according to the
+        passed parameters. Take note of the parameter explanations
+
+        current_password (:obj:`str`, optional):
+            The current password, to authorize changing to ``new_password``
+            Must be set if changing existing 2FA settings.
+            Must not be set if 2FA is currently disabled.
+            Passing this by itself will remove 2FA (if correct)
+
+        new_password (:obj:`str`, optional):
+            The password to set as 2FA. If 2FA was already enabled, ``current_password``
+            must be set. Leaving this blank or ``None`` will remove the password
+
+        hint (:obj:`str`, optional):
+            Hint to be displayed by Telegram when it asks for 2FA.
+            Leaving unspecified is highly discouraged.
+            Has no effect if ``new_password`` is not set
+
+        email (:obj:`str`, optional):
+            Recovery and verification email.
+            Throws ``EmailUnconfirmedError`` if value differs from current one
+            Has no effect if ``new_password`` is not set
+
+        Returns:
+            ``True`` if successful, ``False`` otherwise
+        """
+        if new_password is None and current_password is None:
+            __log__.warn('Both new_password and current_password were not '\
```

Comment:
Not sure if this warning is necessary. No other method shows that on stupid cases.

Reply:
Actually, maybe `raise`'ing some error if trying to clear the password without a current one would be better.

Reply:
manually `raise PasswordEmptyError` then?